File: library-map-frontend/src/types/library.ts
// src/types/library.ts
// Main Library type (the "remodeled model" actually used on the front end)

export interface Library {
  id: number;
  lat: number;
  lon: number;
  GooglePlaceId: string;
  categories?: string;

  name: string;
  address: string;
  websiteUrl?: string;
  websiteUrl2?: string;
  openingHoursJson?: string | null;

}

File: library-map-backend/LibraryMap.Api/Models/Library.cs
namespace LibraryMap.Api.Models
{
    public class Library
    {
        public int Id { get; set; }
        public string? OsmType { get; set; }      // "node" / "way" / "relation"
        public string? OsmId { get; set; }          //way/663054311
        public double Lat { get; set; }
        public double Lon { get; set; }
        public string GooglePlaceId { get; set; } = null!;
        public string? Categories { get; set; }
        public string? Comment { get; set; }


        public string Name { get; set; } = null!;
        public string? Address { get; set; }
        public string? WebsiteUrl { get; set; }
        public string? WebsiteUrl2 { get; set; }
        public string? OpeningHoursJson { get; set; }

    }

}
